### PR TITLE
Migration of the Sonos 2.0 binding from openHAB 2 to Eclipse Smarthome

### PR DESCRIPTION
**Pull request by kgoderis**
Also-by: Andre Fuechsel <a.fuechsel@telekom.de>
Also-by: Dan Cunningham <dan@digitaldan.com>
Signed-off-by: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/binding/binding.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<binding:binding id="sonos"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:binding="http://eclipse.org/smarthome/schemas/binding/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/binding/v1.0.0 http://eclipse.org/smarthome/schemas/binding-1.0.0.xsd">
+
+	<name>Sonos Binding</name>
+	<description>This is the binding for the Sonos multi-room audio system.</description>
+	<author>Karel Goderis</author>
+
+	<config-description>
+		<parameter name="opmlPartnerID" type="text">
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.sonos/ESH-INF/binding/binding.xml]: Please change this as well from partner id to url.

File: distribution/smarthome/conf/smarthome.cfg [hunk 2]
```diff
@@ -18,3 +18,5 @@ org.eclipse.smarthome.threadpool:discovery=3
 
 # Non-scheduled thread pools can also provide a max size
 org.eclipse.smarthome.threadpool:safeCall=3,10
+
+org.eclipse.smarthome.binding.sonos:opmlUrl=http://opml.radiotime.com/Describe.ashx?c=nowplaying&partnerId=IAeIhU42&id=%id&serial=%serial
```
Comment (kaikreuzer) [inline distribution/smarthome/conf/smarthome.cfg]: Please remove this line - this was the whole exercise about the opml url: Not having any external url referenced from Eclipse SmartHome. We will add this setting instead to openHAB 2.

**Comment by kaikreuzer** [comment]
Thanks for this PR, Karel!
Besides my comments above, please also do the following:
- add an about.html to the bundle (see B.2. in https://www.eclipse.org/smarthome/documentation/development/guidelines.html)
- squash your commits into a single one
- add "Also-By" lines to the commit comment for every contributor that contributed to the Sonos binding in the past (e.g. digitaldan), see https://wiki.eclipse.org/Development_Resources/Contributing_via_Git#The_Commit_Record.

**Comment by kaikreuzer** [comment]
@kgoderis, did you see my comments?

**Comment by kgoderis** [comment]
Yes, I did but did not find the time yet to process them

**Comment by kgoderis** [comment]
I think we are ok now

**Comment by kaikreuzer** [comment]
Looks good, thanks a lot!
I have submitted CQ 10204, so now we have to wait for the Eclipse IP team to approve the contribution.

**Comment by kaikreuzer** [comment]
The CQ has been withdrawn and I am waiting for an update of this PR to resolve 3rd party dependency of the XMLParser.

**Comment by kgoderis** [comment]
Will close this one and open from  a newly created branch

**Comment by maggu2810** [comment]
new PR: #548

**Pull request closed without merging**